src/main/java/com/amplitude/Amplitude.java

```diff
@@ -34,31 +53,74 @@ public void init(String key) {
         apiKey = key;
     }
 
+    public void setLogMode(AmplitudeLog.LogMode logMode) {
+        this.logger.setLogMode(logMode);
+    }
+
     public void logEvent(Event event) {
-        try {
-            Future<Void> futureResult = CompletableFuture.supplyAsync(() -> {
-                syncHttpCall(event);
-                return null;
-            });
-            futureResult.get(Constants.NETWORK_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        } catch (ExecutionException e) {
-            e.printStackTrace();
-        } catch (TimeoutException e) {
-            e.printStackTrace();
+        List<Event> listOfOne = new ArrayList<>();
+        listOfOne.add(event);
+        logEvents(listOfOne);
+    }
+
+    public void logEvents(Collection<Event> events) {
+        eventsToSend.addAll(events);
+        if (eventsToSend.size() >= Constants.DEF_EVENT_BUFFER_COUNT) {
+            flushEvents();
+        }
+        else {
+            tryToFlushEventsIfNotFlushing();
         }
     }
 
-    public void setLogMode(AmplitudeLog.LogMode logMode) {
-        this.logger.setLogMode(logMode);
+    private void tryToFlushEventsIfNotFlushing() {
+        if (!aboutToStartFlushing) {
+            aboutToStartFlushing = true;
+            Thread flushThread =
+                    new Thread(() -> {
+                        try {
+                            Thread.sleep(Constants.DEF_EVENT_BUFFER_TIME);
+                        } catch (InterruptedException e) {
+
+                        }
+                        flushEvents();
+                        aboutToStartFlushing = false;
```

Comment:
Will this guaranteed to be called when http call finishes?

Reply:
Actually, this is a good point, this will be called in the middle of an HTTP request. `flushEvents()` does not block the thread. So in theory another timed flush could start, which is OK